### ISSUE TITLE
Upgrade to Rails 6

### ISSUE THREAD

**Issue by gravitystorm** [openstreetmap/openstreetmap-website#2352]
Rails 6.0 [has been released](https://weblog.rubyonrails.org/2019/8/15/Rails-6-0-final-release/), and it contains some things that will be of interest to us:

* Parallel testing - we have a lot of tests!
* Multiple databases - this will allow us to split database read requests into their own connection, and route them to the database replicas. This is something cgimap already does and it will be great to have this here too.
* Action Mailbox - this would be a lower priority, but it will allow us to use standard methods for processing incoming emails. It would be of most use to secondary deployments, who might have less control over receiving their emails than OSMF does.

The major blocker for the upgrade is, as ever, the [`composite_primary_keys`](https://github.com/composite-primary-keys/composite_primary_keys) gem, which needs a 12.x release to support rails 6. We can wait on that, for now.

I've looked through the rest of the [release notes](https://edgeguides.rubyonrails.org/6_0_release_notes.html) and the [upgrade guide](https://edgeguides.rubyonrails.org/upgrading_ruby_on_rails.html#upgrading-from-rails-5-2-to-rails-6-0), and nothing particularly sticks out to me. If there's anything else that will block our upgrade, or any other new feature that we should also take advantage of, add your comment here!

**Comment by tomhughes** [comment]
I started work on this last night ;-)

First signs are good and I got to the point where the majority of tests pass - obviously the big blocker there is CPK as you say but also https://github.com/rails/actionpack-page_caching (believed to be ready but needs a release, https://github.com/rails/actionpack-page_caching/issues/58) and https://github.com/rails/record_tag_helper (also believed to be ready but needs a release, https://github.com/rails/record_tag_helper/pull/10#issuecomment-523135585).

I think CPK is also reasonable shape - there have been a number rc releases during rails 6 devleopment so it probably just needs finishing off.

**Comment by tomhughes** [comment]
So CPK is now available and with that installed and page caching disabled I have most tests passing.

There's a few that are failing on `div_for` and friends (which needs the record_tag_helper update) and also some that seem to be trigger what I think is a bug in rails that I have narrowed down and [reported](https://github.com/rails/rails/issues/37167).

**Comment by gravitystorm** [comment]
Looks like the actionpack-page_caching and record_tag_helper gems are still waiting on releases.
